mhs/common/mhs_common/workflow/tests/test_sync_async.py

```diff
@@ -1,66 +1,78 @@
-        expected_context = {
-            ebxml_envelope.MESSAGE_ID: sentinel.message_id,
-            sync_async.ASYNC_RESPONSE_EXPECTED: True,
-            ebxml_envelope.FROM_PARTY_ID: PARTY_ID,
-            ebxml_envelope.CONVERSATION_ID: MOCK_UUID,
-            ebxml_request_envelope.MESSAGE: sentinel.message
-        }
-        interaction_details = {sync_async.ASYNC_RESPONSE_EXPECTED: True}
-        mock_envelope.return_value.serialize.return_value = \
-            sentinel.ebxml_id, sentinel.http_headers, sentinel.ebxml_message
 
-        is_async, actual_response = self.workflow.prepare_message(interaction_details,
-                                                                  sentinel.message,
-                                                                  sentinel.message_id)
 
-        mock_envelope.assert_called_with(expected_context)
-        self.assertTrue(mock_envelope.return_value.serialize.called)
-        self.assertTrue(is_async)
-        self.assertIs(sentinel.ebxml_message, actual_response)
+class TestSyncAsyncWorkflowOutbound(TestCase):
 
-    def test_prepare_message_sync(self):
-        interaction_details = {sync_async.ASYNC_RESPONSE_EXPECTED: False}
+    def setUp(self):
+        self.persistence = MagicMock()
+        self.work_description = MagicMock()
+        self.work_description_store = MagicMock()
+        self.resync = MagicMock()
+        self.workflow = sync_async.SyncAsyncWorkflow(PARTY_ID,
+                                                     work_description_store=self.work_description_store,
+                                                     resynchroniser=self.resync)
 
-        is_async, actual_response = self.workflow.prepare_message(interaction_details,
-                                                                  sentinel.message,
-                                                                  sentinel.message_id)
+    @patch('mhs_common.state.work_description.create_new_work_description')
+    @test_utilities.async_test
+    async def test_sync_async_happy_path(self, wd_mock):
+        wd_mock.return_value = MagicMock()
+        async_workflow = MagicMock()
+        result = (200, {sync_async.MESSAGE_DATA: "Data", sync_async.CORRELATION_ID: 'cor123'})
+        async_workflow.handle_outbound_message.return_value = test_utilities.awaitable(result)
+
+        await self.workflow.handle_sync_async_outbound_message('id123', 'cor123', {},
+                                                               'payload',
+                                                               async_workflow
+                                                               )
+        wd_mock.assert_called_with(self.work_description_store,
+                                   'id123',
+                                   workflow.SYNC_ASYNC,
+                                   outbound_status=wd.MessageStatus.OUTBOUND_MESSAGE_RECEIVED)
+        async_workflow.handle_outbound_message.assert_called_once()
```

Comment:
We should check the return values here, and also verify that the async workflow is called with the expected arguments.